app/services/notifications/consolidation_planner.rb

```diff
@@ -13,11 +13,15 @@ def consolidate_or_save!(notification)
 
     def plan_for(notification)
       consolidation_plans = [liked_by_two_users, liked, group_message_summary, group_membership]
-      consolidation_plans.concat(DiscoursePluginRegistry.notification_consolidation_plans)
+      consolidation_plans.concat(plugin_consolidation_plans)
 
       consolidation_plans.detect { |plan| plan.can_consolidate_data?(notification) }
     end
 
+    def plugin_consolidation_plans
+      DiscoursePluginRegistry.notification_consolidation_plans.map(&:deep_dup)
```

Comment:
Will we be able to improve the plugin API to instantiate objects instead? Having to use `deep_dup` here feels like we're working around a bug in the API.

Reply:
I agree, but it could be tricky given how things currently work. I'm going to think about this a bit more and see if I can come up with a better solution.

I'm converting this PR to a draft. I'm not aware of this causing any issues yet, so there's no rush in getting it merged.